### PR TITLE
Suppress LineLength for Track 1

### PR DESCRIPTION
**Pull request by conniey**
Suppress LineLength checkstyle rule for Track 1 libraries.

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -115,11 +115,14 @@
...
+  <!-- Suppress LineLength for samples and tests -->
   <suppress checks="LineLength" files=".*[/\\](samples|test)[/\\].*\.java"/>
 
-  <!-- Suppress lineLength for Storage on Preview 3. https://github.com/Azure/azure-sdk-for-java/issues/5125 -->
-  <suppress checks="LineLength" files=".*[/\\]com[/\\]azure[/\\]storage[/\\]*"/>
+  <!-- Suppress LineLength for Storage on Preview 3. https://github.com/Azure/azure-sdk-for-java/issues/5125 -->
+  <suppress checks="LineLength" files=".*[/\\]com[/\\]azure[/\\]storage[/\\]"/>
+
+  <!-- Suppress LineLength for Track 1 libraries. -->
+  <suppress checks="LineLength" files=".*[/\\]com[/\\]microsoft[/\\]"/>
```
Comment (mssfang) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]:
```suggestion
  <suppress checks="LineLength" files=".*[/\\]com[/\\]microsoft[/\\].*"/>
```

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 2]
```diff
@@ -115,11 +115,14 @@
...
   <suppress checks="JavadocPackage" files=".*[/\\](samples|test)[/\\].*\.java"/>
 
-  <!-- Suppress lineLength for samples and tests -->
+  <!-- Suppress LineLength for samples and tests -->
   <suppress checks="LineLength" files=".*[/\\](samples|test)[/\\].*\.java"/>
 
-  <!-- Suppress lineLength for Storage on Preview 3. https://github.com/Azure/azure-sdk-for-java/issues/5125 -->
-  <suppress checks="LineLength" files=".*[/\\]com[/\\]azure[/\\]storage[/\\]*"/>
+  <!-- Suppress LineLength for Storage on Preview 3. https://github.com/Azure/azure-sdk-for-java/issues/5125 -->
+  <suppress checks="LineLength" files=".*[/\\]com[/\\]azure[/\\]storage[/\\]"/>
```
Comment (mssfang) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]:
```suggestion
  <suppress checks="LineLength" files=".*[/\\]com[/\\]azure[/\\]storage[/\\].*"/>
```

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml (hunk 1 above)
Comment (conniey) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: Not required. The regex will match still.

Comment (conniey) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: We just want the regex to result in an isMatch when we evaluate a file name with it. The `.*` isn't needed to basically take anything else.

Comment (conniey) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: To double check, the suppression line here basically does the logic. It looks for any match to that regex, then returns true or false if a match is found.

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressFilterElement.java#L195

**Pull request merged**